aya/src/obj/mod.rs

```diff
@@ -657,10 +672,23 @@ mod tests {
 
     #[test]
     fn test_parse_map_def() {
+        #![allow(unused_variables)] // map_def is used by the assertion
```

Comment:
the compiler is giving you a warning because using `matches!` map_def is not being used, it's being re-bound in the pattern.

If you want to factor out `map_def` - which looks good to me - you need to change the test to use `assert_eq!` instead of `matches!`. Eg:
```suggestion
assert_eq!(parse_map_def("foo", bytes_of(&def)).unwrap(), def)
```

For that to work, you'll have to make `bpf_map_def`  derive `PartialEq`

Reply:
Fixed in 86b639b